### PR TITLE
Signup refactor

### PR DESCRIPTION
**Pull request by tstefanovska**

### CODE REVIEW

File: lib/feature/auth/login/ui/login_view.dart [hunk 1]
```diff
@@ -106,15 +106,17 @@ class LoginView extends StatelessWidget {
...
-    context.read<AuthRouterDelegate>().setSignupNavState();
+    context.read<AuthRouterDelegate>().setSignupUsernameNavState();
   }
 
   onSelected(BuildContext context, String item) {
-    if(item == 'theme'){
-      final themeNotifier = Provider.of<ThemeChangeNotifier>(context, listen: false);
+    if (item == 'theme') {
+      final themeNotifier =
+          Provider.of<ThemeChangeNotifier>(context, listen: false);
```
Comment (dimitar-zabaznoski) [inline lib/feature/auth/login/ui/login_view.dart:115]: can we replace this with context.read< ThemeChangeNotifier >()?

File: android/app/build.gradle [hunk 2]
```diff
@@ -26,8 +26,8 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 //todo uncomment when firebase is configured
-//apply plugin: 'com.google.gms.google-services'
-//apply plugin: 'com.google.firebase.crashlytics'
+apply plugin: 'com.google.gms.google-services'
+apply plugin: 'com.google.firebase.crashlytics'
```
Comment (dimitar-zabaznoski) [inline android/app/build.gradle:30]: why this? this enables firebase

**Review by dimitar-zabaznoski — APPROVED** [review]

File: android/app/build.gradle (hunk 2 above)
Comment (tstefanovska) [reply android/app/build.gradle:30]: Reversed this, thanks! :)

File: lib/feature/auth/login/ui/login_view.dart (hunk 1 above)
Comment (tstefanovska) [reply lib/feature/auth/login/ui/login_view.dart]: Yes this works, I changed it, as well as the localizationNotifier

**Pull request merged**